bundle/API/Repository/Values/Tags/Tag.php

```diff
@@ -33,7 +33,7 @@ final class Tag extends ValueObject
     /**
      * Tag ID.
      */
-    protected int $id;
+    protected ?int $id = null;
```

Comment:
Can you show me a stack trace when this field is accessed before initialization? Rather than setting this field like this, we should fix the error where it happens.

Reply:
Hello,

sorry for the lack of informations.

Error:
Typed property Netgen\TagsBundle\API\Repository\Values\Tags\Tag::$id must not be accessed before initialization

  at vendor/ibexa/core/src/contracts/Repository/Values/ValueObject.php:98
  at Ibexa\Contracts\Core\Repository\Values\ValueObject->__get('id')
     (vendor/netgen/tagsbundle/bundle/API/Repository/Values/Tags/Tag.php:127)
  at Netgen\TagsBundle\API\Repository\Values\Tags\Tag->__get('id')
     (vendor/netgen/tagsbundle/bundle/Core/FieldType/Tags/Type.php:323)
  at Netgen\TagsBundle\Core\FieldType\Tags\Type->validate(object(FieldDefinition), object(Value))
     (vendor/ibexa/core/src/lib/Repository/Values/ContentType/FieldType.php:297)
  at Ibexa\Core\Repository\Values\ContentType\FieldType->validateValue(object(FieldDefinition), object(Value))
     (vendor/ibexa/content-forms/src/lib/Validator/Constraints/FieldValueValidator.php:52)
  at Ibexa\ContentForms\Validator\Constraints\FieldValueValidator->validate(object(FieldData), object(FieldValue))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:759)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateInGroup(object(FieldData), '00000000000011f90000000000000000', object(ClassMetadata), 'Default', object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:488)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateClassNode(object(FieldData), '00000000000011f90000000000000000', object(ClassMetadata), 'data.fieldsData[tag]', array('Default'), null, 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:313)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateObject(object(FieldData), 'data.fieldsData[tag]', array('Default'), 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:370)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateEachObjectIn(array('page' => object(FieldData), 'title_bo' => object(FieldData), 'title' => object(FieldData), 'publication_dt' => object(FieldData), 'featured_image' => object(FieldData), 'short_description' => object(FieldData), 'seo_metadatas' => object(FieldData), 'tag' => object(FieldData)), 'data.fieldsData', array('Default'), object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:646)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateGenericNode(array('page' => object(FieldData), 'title_bo' => object(FieldData), 'title' => object(FieldData), 'publication_dt' => object(FieldData), 'featured_image' => object(FieldData), 'short_description' => object(FieldData), 'seo_metadatas' => object(FieldData), 'tag' => object(FieldData)), object(ContentUpdateData), '00000000000011ef0000000000000000:Ibexa\\ContentForms\\Data\\Content\\ContentUpdateData:fieldsData', object(PropertyMetadata), 'data.fieldsData', array('Default'), array('Default'), 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:516)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateClassNode(object(ContentUpdateData), '00000000000011ef0000000000000000', object(ClassMetadata), 'data', array('Default'), null, 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:313)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateObject(object(ContentUpdateData), 'data', array('Default'), 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:138)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validate(object(ContentUpdateData), null, array('Default'))
     (vendor/symfony/form/Extension/Validator/Constraints/FormValidator.php:108)
  at Symfony\Component\Form\Extension\Validator\Constraints\FormValidator->validate(object(Form), object(Form))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:759)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateInGroup(object(Form), '00000000000012770000000000000000', object(ClassMetadata), 'Default', object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:488)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateClassNode(object(Form), '00000000000012770000000000000000', object(ClassMetadata), '', array('Default'), null, 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:313)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateObject(object(Form), '', array('Default'), 1, object(ExecutionContext))
     (vendor/symfony/validator/Validator/RecursiveContextualValidator.php:138)
  at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validate(object(Form), null, array('Default'))
     (vendor/symfony/validator/Validator/RecursiveValidator.php:93)
  at Symfony\Component\Validator\Validator\RecursiveValidator->validate(object(Form), null, null)
     (vendor/symfony/validator/Validator/TraceableValidator.php:66)
  at Symfony\Component\Validator\Validator\TraceableValidator->validate(object(Form))
     (vendor/symfony/form/Extension/Validator/EventListener/ValidationListener.php:50)
  at Symfony\Component\Form\Extension\Validator\EventListener\ValidationListener->validateForm(object(PostSubmitEvent), 'form.post_submit', object(EventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure), object(Closure)), 'form.post_submit', object(PostSubmitEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(PostSubmitEvent), 'form.post_submit')
     (vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php:33)
  at Symfony\Component\EventDispatcher\ImmutableEventDispatcher->dispatch(object(PostSubmitEvent), 'form.post_submit')
     (vendor/symfony/form/Form.php:681)
  at Symfony\Component\Form\Form->submit(array(), true)
     (vendor/symfony/form/Extension/HttpFoundation/HttpFoundationRequestHandler.php:109)
  at Symfony\Component\Form\Extension\HttpFoundation\HttpFoundationRequestHandler->handleRequest(object(Form), object(Request))
     (vendor/symfony/form/Form.php:503)
  at Symfony\Component\Form\Form->handleRequest(object(Request))
     (vendor/ibexa/content-forms/src/lib/Content/View/Filter/ContentEditViewFilter.php:114)
  at Ibexa\ContentForms\Content\View\Filter\ContentEditViewFilter->handleContentEditForm(object(FilterViewBuilderParametersEvent), 'view.builder_parameter_collection', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(FilterViewBuilderParametersEvent), 'view.builder_parameter_collection', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'view.builder_parameter_collection', object(FilterViewBuilderParametersEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(FilterViewBuilderParametersEvent), 'view.builder_parameter_collection')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:154)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(FilterViewBuilderParametersEvent), 'view.builder_parameter_collection')
     (vendor/ibexa/core/src/bundle/Core/EventListener/ViewControllerListener.php:68)
  at Ibexa\Bundle\Core\EventListener\ViewControllerListener->getController(object(ControllerEvent), 'kernel.controller', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ControllerEvent), 'kernel.controller', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.controller', object(ControllerEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(ControllerEvent), 'kernel.controller')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:154)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(ControllerEvent), 'kernel.controller')
     (vendor/symfony/http-kernel/HttpKernel.php:140)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (vendor/autoload_runtime.php:35)
  at require_once('/srv/apps/cms/vendor/autoload_runtime.php')
     (public/index.php:5)

Reply:
Okay, I get it, you can fix this with:

```
if (!isset($tag->id) || $tag->id < 1) {
```

in `bundle/Core/FieldType/Tags/Type.php` on line 324.

Reply:
````
if (!isset($tag->id) || $tag->id < 1) {
````
Its not working.

https://stackoverflow.com/questions/59265625/why-i-am-suddenly-getting-a-typed-property-must-not-be-accessed-before-initiali

That's why I provided this fix.

Reply:
Changing the field type should also not be done here. If `if (!isset($tag->id) || $tag->id < 1) {` does not work, I'll try to find out why.